tests/testthat/test-parsnip-survival-censoring-weights.R

```diff
@@ -1,37 +1,122 @@
-test_that('calculate weight time', {
+test_that("graf_weight_time_vec() calculates weight time", {
   skip_if_not_installed("survival")
 
-  times <- 1:10
-  cens <- rep(0:1, times = 5)
+  eval_time_10 <- 10
 
-  surv_obj <- survival::Surv(times, cens)
-  n <- length(surv_obj)
+  # Graf et al (1999) Category 1
+  event_before_or_at_eval_time <- survival::Surv(
+    time = c(eval_time_10 - 1, eval_time_10),
+    event = c(1, 1)
+  )
+  expect_equal(
+    parsnip:::graf_weight_time_vec(event_before_or_at_eval_time, eval_time = eval_time_10) %>%
+      unname(),
```

Comment:
I think we can drop the names inside of `parsnip:::graf_weight_time_vec()` but thought that that's better to do that in a separate PR pair.

Reply:
Update: Looks like this is not named anymore:

``` r
eval_time_10 <- 10

# Graf et al (1999) Category 1
event_before_or_at_eval_time <- survival::Surv(
  time = c(eval_time_10 - 1, eval_time_10),
  event = c(1, 1)
)

parsnip:::graf_weight_time_vec(event_before_or_at_eval_time, eval_time = eval_time_10)
#> [1]  9 10
```

<sup>Created on 2023-11-09 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Reply:
TODO for myself: remove the `unname()` from the test

Reply:
resolved in [8fafcc4](https://github.com/tidymodels/extratests/pull/98/commits/8fafcc421fadb65a41038af547b2dc362b542223)